security-force-monitor/sfm-cms
# use Units map code and upgrade django to 3.2.13

# Overview
Fixes:
- #804
- bumps Django to 3.2.13 due to [a security advisory](https://www.djangoproject.com/weblog/2022/apr/11/security-releases/)

Dependent on merging [the PR for the complex_fields dependency](https://github.com/security-force-monitor/complex_fields/pull/2).

## Notes
The code was expecting incident locations to have x,y coordinates, but sometimes the data is a polygon geometry type.

I used [the code from this PR](https://github.com/security-force-monitor/sfm-cms/pull/728) to render the map depending on geometry type. Very easy to work with!

## Testing
- Pull down branch
- Import `ye`: `docker-compose --env-file .env.import run --rm app make -e ye_cc_import`
- For comparison, import another country (small dataset): `docker-compose --env-file .env.import run --rm app make -e mr_cc_import`
- See a location with a polygon (in ye data): http://localhost:8000/en/violation/view/045cab2d-fe37-4487-b364-b7b202529c62/
- See a location with a point (in mr data): http://localhost:8000/en/violation/view/f19204ab-301a-4a77-9c6d-a05dedc09472/

Note that with the partial import, an incident is missing the country. This happens to me with both the master branch and this PR's code change.
<img width="456" alt="Screen Shot 2022-04-26 at 2 09 21 PM" src="https://user-images.githubusercontent.com/38969506/165374659-00e3ed39-167b-40df-af21-ea6c8af7aa0d.png">
